Crew manage common crewnamelist

zappyboy7 opened the pull request:
feat : crewManage 기본형식 및 리스트 기능

zappyboy7 committed:
feat : crew manage 기본 형식 및 리스트 기능

hackey1318 commented on runnerSquare/src/main/java/com/ict/rs/controller/CrewManageController.java:
```diff
@@ -0,0 +1,53 @@
+	@RequestMapping("/crew/crewManage")
+    public String crewManage(Model model) {
+		
+        // �ʿ��� �����͸� �𵨿� �߰�
```
한글 깨짐 잡아주세요.

hackey1318 commented on runnerSquare/src/main/java/com/ict/rs/controller/CrewManageController.java:
```diff
@@ -0,0 +1,53 @@
+	@ResponseBody
+	@GetMapping("/crew/crewManage/memberList")
+	public List<CrewManageVO> crewMemberList(@RequestParam Integer userNo){
+		System.out.print("����Ȯ��"+userNo);
```
Controller에 @Slf4j를 이용하여 log를 찍는 형태로 해주세요. System.out을 이용해서 서버에 정보를 Write하면 이 또한 작업이기에 서버 자원을 사용하는 것입니다.

hackey1318 commented on runnerSquare/src/main/java/com/ict/rs/controller/CrewManageController.java:
```diff
@@ -0,0 +1,53 @@
+	public List<CrewManageVO> crewMemberList(@RequestParam Integer userNo){
+		System.out.print("����Ȯ��"+userNo);
+		List<CrewManageVO> result = service.crewMemberList(userNo);
+		System.out.print(result.size());
```
이 또한 확인용인 것은 지우고 올려주세요.

hackey1318 commented on runnerSquare/src/main/java/com/ict/rs/service/CrewManageServiceImpl.java:
```diff
@@ -0,0 +1,30 @@
+
+	@Override
+	public List<CrewManageVO> crewMemberList(int userNo) {
+		// TODO Auto-generated method stub
```
불필요한 주석 지워주세요.

hackey1318 commented on runnerSquare/src/main/webapp/WEB-INF/views/crew/crewManage.jsp:
```diff
@@ -0,0 +1,282 @@
+
+function loadCrewNameList() {
+	
+	const params = { userNo: 4 };
```
하드코딩 된 값을 이용한 경우 주석으로 추후 변경을 어떻게 할 것이다는 것도 기입해주면 좋습니다.

hackey1318 commented on runnerSquare/src/main/resources/mapper/CrewManageMapper.xml:
```diff
@@ -0,0 +1,49 @@
+  "https://mybatis.org/dtd/mybatis-3-mapper.dtd">
+  
+<mapper namespace="com.ict.rs.dao.CrewManageDAO">
+	<select id="crewNameList" resultType="com.ict.rs.vo.CrewManageVO">
```
해당 쿼리가 뭘 원하는지를 알고 작성하면 좋을거 같아요.

> SELECT
	u.nickName,	// 사용자 NickName
	cm.role,		// 소속 크루에서 사용자 권한
	cm.status,		// 소속 크루 상태
	cm.createdAt,	// 소속 크루 가입일
	cm.updatedAt,	// 소속 크루에서 정보 변경일(상태 변경, 권한 변경)
	c.no,		//  소속 크루 PK
	c.name,		// 소속 크루명
	c.city,			// 활동 시/도
	c.district,		// 활동 시/군/구
	c.runningDay,	// 소속 크루 러닝데이
	c.status		// 소속 크루 상태
FROM users AS u
LEFT JOIN crewMembers AS cm ON u.no = cm.userNo
LEFT JOIN crew AS c ON cm.crewNo = c.no
WHERE u.id = ${userId};

단순히 봤을 때, 이거 말고 정보들이 더 필요한가?
그리고 해당 페이지는 결국 로그인 한 유저가 들어와서 보는 페이지이기에 NickName정보가 굳이 필요할까??도 고민하면 좋아.

hackey1318 commented on runnerSquare/src/main/resources/mapper/CrewManageMapper.xml:
```diff
@@ -0,0 +1,49 @@
+		where userNo=${param1};
+	</select>
+	
+	<select id="crewMemberList" resultType="com.ict.rs.vo.CrewManageVO">
```
이 또한 크루의 맴버를 조회하는 것이기에 처음에 크루 조회했을 때 크루 no정보를 가지고 있다면 크루원에서 crewNo를 이용해서 사용자들을 조회하는 형태가 되어야한다고 생각해
지금 보았을 때, 정말 불필요한 정보들도 다 가지고 가고 있어서 수정했으면 함

hackey1318 reviewed (commented):
리뷰관련 수정 및 정보 확인 해주세요

zappyboy7 committed:
develop pull 후 충돌 해결

zappyboy7 committed:
feat :
crewManage-ui에서 crewManage-commonlist로 브렌치변경
1. 한글 깨짐현상 fix(git-config파일에 utf-8적용)
2. mapper 수정
ㅇ 필요한 데이터만 가져오게끔 쿼리문수정
ㅇ 활동지역에서 시, 시/군/구로 수정
ㅇ 이에따른 crewManageVO 수정
3.필요없는 주석문 삭제
4.테스트를위한 system.out삭제
5.crewManage.jsp하드코딩된 값 주석 추가
ㅇ 추후 수정필요

zappyboy7 committed:
Merge branch 'crewManage-common' of https://github.com/hackey1318/ICT-Mini-Project-1 into crewManage-common-crewnamelist

# Conflicts:
#	runnerSquare/src/main/webapp/resources/header.jspf
